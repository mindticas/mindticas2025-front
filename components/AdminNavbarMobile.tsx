import { AdminNavbarMobileProps } from '@/interfaces/navItems/navItems';
import { handleRefresh } from '@/services/RefreshToken';
import {
    Box,
    Flex,
    Link as ChakraLink,
    IconButton,
    Stack,
    Button,
    Text,
    Image,
} from '@chakra-ui/react';
import { LogOut, Menu, Scissors } from 'lucide-react';
import NextLink from 'next/link';
import CalendarImg from '@/public/google-calendar.png';

export default function AdminNavbarMobile({
    navItems,
    pathname,
    isMobileMenuOpen,
    onToggleMenu,
}: AdminNavbarMobileProps) {
    return (
        <Box bg='gray.800' color='white' shadow='lg'>
            <Flex
                maxW='7xl'
                mx='auto'
                px={{ base: 4, md: 6, lg: 8 }}
                h='16'
                align='center'
                justify='space-between'
            >
                {/* Logo */}
                <ChakraLink
                    as={NextLink}
                    href='/admin/citas'
                    display='flex'
                    alignItems='center'
                    _hover={{ textDecoration: 'none' }}
                    color='white'
                >
                    <Scissors size={32} />
                    <Text ml='2' fontSize='xl' fontWeight='bold'>
                        Elegangster Admin
                    </Text>
                </ChakraLink>
                <IconButton
                    aria-label='Abrir menÃº'
                    variant='ghost'
                    size='2xl'
                    onClick={onToggleMenu}
                >
                    <Menu color='white' />
                </IconButton>
            </Flex>
            {/* Mobile menu */}
            {isMobileMenuOpen && (
                <Box pb='4' display={{ xl: 'none' }}>
                    <Stack px='2'>
                        {navItems.map((item) => {
                            const Icon = item.icon;
                            const active = pathname === item.href;
                            return (
                                <ChakraLink
                                    key={item.name}
                                    as={NextLink}
                                    href={item.href}
                                    px='3'
                                    py='2'
                                    rounded='md'
                                    fontSize='lg'
                                    fontWeight='medium'
                                    bg={active ? 'gray.900' : 'transparent'}
                                    _hover={{
                                        bg: 'gray.700',
                                        textDecoration: 'none',
                                    }}
                                    display='flex'
                                    alignItems='center'
                                    onClick={onToggleMenu}
                                    color='white'
                                >
                                    <Icon
                                        width={20}
                                        height={20}
                                        style={{ marginRight: '8px' }}
                                    />
                                    {item.name}
                                </ChakraLink>
                            );
                        })}
                        <Button
                            variant='ghost'
                            px='3'
                            py='2'
                            justifyContent='flex-start'
                            fontSize='lg'
                            fontWeight='medium'
                            color='gray.300'
                            _hover={{ bg: 'gray.700', color: 'white' }}
                        >
                            <LogOut size={16} />
                            Salir
                        </Button>
                        <Button
                            onClick={handleRefresh}
                            justifyContent='flex-start'
                            variant='ghost'
                            px='3'
                            py='2'
                            fontSize='lg'
                            fontWeight='medium'
                            color='gray.300'
                            _hover={{ bg: 'gray.700', color: 'white' }}
                        >
                            <Image
                                boxSize='25px'
                                rounded='md'
                                src={CalendarImg.src}
                            />
                            <Text fontSize='md' fontWeight='medium'>
                                Sincronizar
                            </Text>
                        </Button>
                    </Stack>
                </Box>
            )}
        </Box>
    );
}
