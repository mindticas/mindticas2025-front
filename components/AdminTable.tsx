'use client';

import { Table, Box, Spinner } from '@chakra-ui/react';
import { ReactNode } from 'react';

interface ColumnDef<T> {
    key: string;
    header: string;
    render: (item: T) => ReactNode;
    align?: 'left' | 'center' | 'right';
    width?: string | number;
}

interface AdminTableProps<T> {
    data: T[];
    columns: ColumnDef<T>[];
    isLoading?: boolean;
    emptyMessage?: string;
    scrollable?: boolean;
}

export function AdminTable<T>({
    data,
    columns,
    isLoading = false,
    emptyMessage = 'No hay datos disponibles',
}: AdminTableProps<T>) {
    if (isLoading) {
        return (
            <Box
                display='flex'
                alignItems='center'
                justifyContent='center'
                height='300px'
            >
                <Spinner size='xl' />
            </Box>
        );
    }

    const TableContent = (
        <Table.Root size='sm' variant='outline' showColumnBorder>
            <Table.Header>
                <Table.Row bg='#F3F4F6'>
                    {columns.map((column) => (
                        <Table.ColumnHeader
                            key={column.key}
                            textAlign={column.align || 'center'}
                            width={column.width}
                            py={2}
                            fontWeight='bold'
                        >
                            {column.header}
                        </Table.ColumnHeader>
                    ))}
                </Table.Row>
            </Table.Header>
            <Table.Body>
                {data.length === 0 ? (
                    <Table.Row>
                        <Table.Cell
                            p={3}
                            colSpan={columns.length}
                            textAlign='center'
                        >
                            {emptyMessage}
                        </Table.Cell>
                    </Table.Row>
                ) : (
                    data.map((item, index) => (
                        <Table.Row
                            className='tr-table'
                            key={index}
                            borderBottomWidth={1}
                        >
                            {columns.map((column) => (
                                <Table.Cell
                                    key={`${column.key}-${index}`}
                                    textAlign={column.align || 'center'}
                                    px={2}
                                    width='fit-content'
                                >
                                    {column.render(item)}
                                </Table.Cell>
                            ))}
                        </Table.Row>
                    ))
                )}
            </Table.Body>
        </Table.Root>
    );

    return (
        <Table.ScrollArea borderWidth='1px' borderRadius='lg'>
            {TableContent}
        </Table.ScrollArea>
    );
}
