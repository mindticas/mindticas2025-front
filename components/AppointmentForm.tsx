'use client';
import {
    Button,
    Fieldset,
    Flex,
    Input,
    Spinner,
    Stack,
} from '@chakra-ui/react';
import { Field } from '@/components/ui/field';
import React, { useState } from 'react';
import PhoneInput, { Value } from 'react-phone-number-input';
import { isValidPhoneNumber } from 'react-phone-number-input';
import 'react-phone-number-input/style.css';
import { useBookingContext } from '@/context/BookingContext';
import { toaster } from './ui/toaster';
import { createAppointment } from '@/services/AppointmentService';

interface AppointmentFormProps {
    onSuccess: () => void;
    onError: () => void;
}

export default function AppointmentForm({
    onSuccess,
    onError,
}: AppointmentFormProps) {
    const {
        treatment,
        personData,
        setPersonData,
        dateTime,
        setIsSuccessfullyBooked,
    } = useBookingContext();

    // State for loading state
    const [isLoading, setIsLoading] = useState(false);
    // State for phone number validation
    const [isPhoneValid, setIsPhoneValid] = useState(true);

    // Handle phone number change
    const handlePhoneChange = (phone: Value) => {
        setPersonData({
            ...personData,
            phone: phone ? phone.toString() : '',
        });

        // Validate phone number
        if (phone) {
            setIsPhoneValid(isValidPhoneNumber(phone));
        } else {
            setIsPhoneValid(true); // Reset validation if empty
        }
    };

    // Handle input changes and update state
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setPersonData({ ...personData, [name]: value });
    };

    // Function to remove country code (+52) and keep only 10 digits
    const formatPhoneNumber = (phone: string) => {
        if (phone.startsWith('+52')) {
            return phone.slice(3); // Remove +52
        }
        return phone; // If no country code, return as is
    };

    // Handle form submission
    const handleSubmit = async (e: React.FormEvent) => {
        e.preventDefault();

        // Validate phone number before submission
        if (!personData.phone || !isValidPhoneNumber(personData.phone)) {
            toaster.create({
                type: 'error',
                duration: 5000,
                title: 'Número de teléfono inválido o vacío',
            });
            return;
        }

        // Validate form fields before submission
        if (!personData.name || !personData.lastName) {
            onError();
            return;
        }

        // Set loading state spinner
        setIsLoading(true);

        try {
            // Format phone number to remove country code
            const formattedPhone = formatPhoneNumber(personData.phone);
            // Create object with appointment data
            const newAppointment = {
                name: `${personData.name} ${personData.lastName}`,
                phone: formattedPhone,
                treatment_ids: [treatment],
                scheduledStart: dateTime,
            };

            // create appointment in the backend
            await createAppointment(newAppointment);

            // if appointment is created successfully, show success message
            setIsLoading(false);
            onSuccess();
            setIsSuccessfullyBooked(true);
            setPersonData({ name: '', lastName: '', phone: '' });
        } catch (error) {
            setIsLoading(false);
            onError();
        }
    };

    return (
        <form onSubmit={handleSubmit} style={{ width: '100%' }}>
            <Flex p='6' justify='center'>
                <Fieldset.Root
                    borderRadius='lg'
                    boxShadow='md'
                    borderWidth='1px'
                    borderColor='gray.200'
                    size='lg'
                    maxW='sm'
                    p={6}
                >
                    <Stack>
                        <Fieldset.Legend
                            fontSize='lg'
                            fontWeight='600'
                            color='gray.900'
                        >
                            Datos del contacto
                        </Fieldset.Legend>
                        <Fieldset.HelperText color='gray.600' pb={3}>
                            Ingrese los datos del contacto para agendar una cita
                        </Fieldset.HelperText>
                    </Stack>
                    <Fieldset.Content>
                        <Field required label='Nombre'>
                            <Input
                                size='lg'
                                p={1}
                                type='text'
                                name='name'
                                placeholder='Juan'
                                value={personData.name}
                                onChange={handleChange}
                            />
                        </Field>
                        <Field required label='Apellido'>
                            <Input
                                size='lg'
                                p={1}
                                name='lastName'
                                type='text'
                                placeholder='Torres'
                                value={personData.lastName}
                                onChange={handleChange}
                            />
                        </Field>
                        <Field required label='Número de teléfono'>
                            <PhoneInput
                                className='PhoneInputInput'
                                countries={['MX']}
                                defaultCountry='MX'
                                placeholder='Ingresa tu número de teléfono'
                                value={personData.phone}
                                onChange={handlePhoneChange}
                                style={{
                                    padding: '8px',
                                    fontSize: '16px',
                                    border: isPhoneValid
                                        ? '1px solid #ccc'
                                        : '1px solid red',
                                    borderRadius: '4px',
                                    width: '100%',
                                }}
                            />
                            {!isPhoneValid && (
                                <p style={{ color: 'red', fontSize: '14px' }}>
                                    Número de teléfono inválido.
                                </p>
                            )}
                        </Field>
                    </Fieldset.Content>
                    <Button
                        bg='black'
                        color='white'
                        type='submit'
                        w='full'
                        mt={6}
                        disabled={isLoading}
                    >
                        {isLoading ? <Spinner size='sm' mr={2} /> : 'Agendar'}
                    </Button>
                </Fieldset.Root>
            </Flex>
        </form>
    );
}
