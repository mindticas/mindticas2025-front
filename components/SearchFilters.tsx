'use client';

import { Button, Grid, HStack, Input, Menu, Text } from '@chakra-ui/react';
import { SelectRoot, createListCollection } from '@chakra-ui/react';
import {
    SelectContent,
    SelectItem,
    SelectTrigger,
} from '@/components/ui/select';
import {
    ArrowDown,
    ArrowUp,
    CalendarCheck,
    ChevronDown,
    CircleCheckBig,
    CircleX,
    Clock,
} from 'lucide-react';

export interface AppointmentsFilters {
    name: string;
    treatments?: string;
    date?: string;
    status?: string;
    sort?: SortCriteria;
}

export type SortCriteria =
    | 'SERVICE_COUNT_ASC'
    | 'SERVICE_COUNT_DESC'
    | 'CUSTOMER_NAME_ASC'
    | 'CUSTOMER_NAME_DESC'
    | '';

interface SearchFiltersProps {
    filters: AppointmentsFilters;
    onFilterChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onStatusChange?: (status: string) => void;
    onSortChange?: (sort: SortCriteria) => void;
    showStatusFilter?: boolean;
    showDateFilter?: boolean;
    showTreatmentFilter?: boolean;
    showSort?: boolean;
}

export const statusOptions = [
    {
        value: 'confirmed',
        label: 'Confirmado',
        color: 'blue',
        icon: <CalendarCheck size={16} />,
    },
    {
        value: 'pending',
        label: 'Pendiente',
        color: 'yellow',
        icon: <Clock size={16} strokeWidth={3} />,
    },
    {
        value: 'canceled',
        label: 'Cancelado',
        color: 'red',
        icon: <CircleX size={16} strokeWidth={3} />,
    },
    {
        value: 'completed',
        label: 'Completado',
        color: 'green',
        icon: <CircleCheckBig size={16} strokeWidth={3} />,
    },
    { value: '', label: 'Todos los estatus', color: 'gray' },
];

const sortOptions = [
    {
        value: 'SERVICE_COUNT_ASC',
        label: 'Núm. Servicios (menos a más)',
        icon: <ArrowUp size={20} />,
    },
    {
        value: 'SERVICE_COUNT_DESC',
        label: 'Núm. Servicios (más a menos)',
        icon: <ArrowDown size={20} />,
    },
    {
        value: 'CUSTOMER_NAME_ASC',
        label: 'Nombre (A-Z)',
        icon: <ArrowUp size={20} />,
    },
    {
        value: 'CUSTOMER_NAME_DESC',
        label: 'Nombre (Z-A)',
        icon: <ArrowDown size={20} />,
    },
];

/**
 * A functional component that renders a set of search filters for filtering data.
 * The component supports filters for name, treatments, date, and status, with
 * optional visibility for each filter type.
 *
 * @param {SearchFiltersProps} props - The props for the `SearchFilters` component.
 * @param {Object} props.filters - The current filter values.
 * @param {string} props.filters.name - The filter value for the name input.
 * @param {string} [props.filters.treatments] - The filter value for the treatments input.
 * @param {string} [props.filters.date] - The filter value for the date input.
 * @param {string} [props.filters.status] - The filter value for the status dropdown.
 * @param {(event: React.ChangeEvent<HTMLInputElement>) => void} props.onFilterChange - Callback function triggered when a filter input value changes.
 * @param {(value: string) => void} [props.onStatusChange] - Callback function triggered when the status filter value changes.
 * @param {boolean} [props.showStatusFilter=false] - Whether to show the status filter dropdown.
 * @param {boolean} [props.showDateFilter=false] - Whether to show the date filter input.
 * @param {boolean} [props.showTreatmentFilter=false] - Whether to show the treatments filter input.
 *
 * @returns {JSX.Element} The rendered search filters component.
 */
export function SearchFilters({
    filters,
    onFilterChange,
    onStatusChange,
    onSortChange,
    showStatusFilter = false,
    showDateFilter = false,
    showTreatmentFilter = false,
    showSort = false,
}: SearchFiltersProps) {
    const activeSortLabel = sortOptions.find(
        (opt) => opt.value === filters.sort,
    )?.label;

    return (
        <Grid
            templateColumns={{
                base: '1fr',
                md: showSort ? 'repeat(3, 1fr)' : 'repeat(4, 1fr)',
            }}
            gap={4}
            mb={4}
        >
            <Input
                p={2}
                name='name'
                placeholder='Filtrar por nombre'
                fontSize='md'
                value={filters.name}
                onChange={onFilterChange}
            />

            {showTreatmentFilter && (
                <Input
                    p={2}
                    fontSize='md'
                    name='treatments'
                    placeholder='Filtrar por servicio'
                    value={filters.treatments || ''}
                    onChange={onFilterChange}
                />
            )}

            {showDateFilter && (
                <Input
                    p={2}
                    type='date'
                    _dark={{
                        '&::-webkit-calendar-picker-indicator': {
                            filter: 'invert(1)',
                        },
                    }}
                    name='date'
                    fontSize='md'
                    placeholder='Filtrar por fecha'
                    value={filters.date || ''}
                    onChange={onFilterChange}
                />
            )}

            {showStatusFilter && onStatusChange && (
                <SelectRoot
                    name='status'
                    collection={createListCollection({
                        items: statusOptions,
                    })}
                    onValueChange={(e) => onStatusChange(e.value.toString())}
                >
                    <SelectTrigger>
                        {filters.status
                            ? statusOptions.find(
                                  (opt) => opt.value === filters.status,
                              )?.label
                            : 'Filtrar por estado'}
                    </SelectTrigger>
                    <SelectContent backgroundColor='white'>
                        {statusOptions.map(({ value, label }) => (
                            <SelectItem
                                cursor='pointer'
                                _hover={{ backgroundColor: 'gray.100' }}
                                backgroundColor='white'
                                item={value}
                                key={value}
                                p={2}
                            >
                                {label}
                            </SelectItem>
                        ))}
                    </SelectContent>
                </SelectRoot>
            )}

            {/* Sort options */}
            {showSort && onSortChange && (
                <Menu.Root>
                    <Menu.Trigger asChild>
                        <Button variant='subtle' size='sm'>
                            {filters.sort ? (
                                filters.sort.includes('_ASC') ? (
                                    <ArrowUp size={16} />
                                ) : (
                                    <ArrowDown size={16} />
                                )
                            ) : (
                                <ChevronDown size={16} />
                            )}

                            {filters.sort
                                ? `Orden: ${activeSortLabel?.split(' ')[1]}`
                                : 'Ordenar por'}
                        </Button>
                    </Menu.Trigger>
                    <Menu.Positioner>
                        <Menu.Content>
                            {sortOptions.map((option) => (
                                <Menu.Item
                                    key={option.value}
                                    value={option.value}
                                    onClick={() =>
                                        onSortChange(
                                            option.value as SortCriteria,
                                        )
                                    }
                                    _hover={{ bg: 'gray.100' }}
                                >
                                    <HStack
                                        p={2}
                                        width={'100%'}
                                        justifyContent={'space-between'}
                                    >
                                        <Text>{option.label}</Text>
                                        {option?.icon}
                                    </HStack>
                                </Menu.Item>
                            ))}
                        </Menu.Content>
                    </Menu.Positioner>
                </Menu.Root>
            )}
        </Grid>
    );
}
