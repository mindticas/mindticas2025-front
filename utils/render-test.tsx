import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
