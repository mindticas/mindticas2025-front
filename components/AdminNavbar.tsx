'use client';
import { navItems } from '@/utils/navItems';
import { useBreakpointValue } from '@chakra-ui/react';
import { usePathname } from 'next/navigation';
import React, { useState } from 'react';
import AdminNavbarDesktop from './AdminNavbarDesktop';
import AdminNavbarMobile from './AdminNavbarMobile';

export default function AdminNavbar() {
    const pathname = usePathname();
    const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
    const isDesktop = useBreakpointValue({ base: false, md: true });

    const handleToggleMenu = () => setIsMobileMenuOpen((prev) => !prev);

    return isDesktop ? (
        <AdminNavbarDesktop navItems={navItems} pathname={pathname} />
    ) : (
        <AdminNavbarMobile
            navItems={navItems}
            pathname={pathname}
            isMobileMenuOpen={isMobileMenuOpen}
            onToggleMenu={handleToggleMenu}
        />
    );
}
