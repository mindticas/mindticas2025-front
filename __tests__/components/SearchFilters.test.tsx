import { SearchFilters, statusOptions } from '@/components/SearchFilters';
import { render } from '@/utils/render-test';

describe('SearchFilters', () => {
    const mockFilters = {
        name: '',
        treatments: '',
        date: '',
        status: '',
    };

    const mockOnFilterChange = jest.fn();
    const mockOnStatusChange = jest.fn();

    it('should render basic filters correctly', () => {
        const { container } = render(
            <SearchFilters
                filters={mockFilters}
                onFilterChange={mockOnFilterChange}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render with all filters enabled', () => {
        const { container } = render(
            <SearchFilters
                filters={mockFilters}
                onFilterChange={mockOnFilterChange}
                onStatusChange={mockOnStatusChange}
                showStatusFilter={true}
                showDateFilter={true}
                showTreatmentFilter={true}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render with status filter only', () => {
        const { container } = render(
            <SearchFilters
                filters={{ ...mockFilters, status: 'confirmed' }}
                onFilterChange={mockOnFilterChange}
                onStatusChange={mockOnStatusChange}
                showStatusFilter={true}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render with date filter only', () => {
        const { container } = render(
            <SearchFilters
                filters={{ ...mockFilters, date: '2023-01-01' }}
                onFilterChange={mockOnFilterChange}
                showDateFilter={true}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render with treatment filter only', () => {
        const { container } = render(
            <SearchFilters
                filters={{ ...mockFilters, treatments: 'Haircut' }}
                onFilterChange={mockOnFilterChange}
                showTreatmentFilter={true}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render with name filter filled', () => {
        const { container } = render(
            <SearchFilters
                filters={{ ...mockFilters, name: 'John Doe' }}
                onFilterChange={mockOnFilterChange}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    it('should render status options correctly', () => {
        // Test that all status options are properly exported
        expect(statusOptions).toMatchSnapshot();
    });
});
