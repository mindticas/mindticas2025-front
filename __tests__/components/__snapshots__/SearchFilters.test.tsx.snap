// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilters should render basic filters correctly 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value=""
    />
  </div>
</div>
`;

exports[`SearchFilters should render status options correctly 1`] = `
[
  {
    "color": "blue",
    "icon": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {
        "size": 16,
      },
      "type": {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "label": "Confirmado",
    "value": "confirmed",
  },
  {
    "color": "yellow",
    "icon": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {
        "size": 16,
        "strokeWidth": 3,
      },
      "type": {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "label": "Pendiente",
    "value": "pending",
  },
  {
    "color": "red",
    "icon": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {
        "size": 16,
        "strokeWidth": 3,
      },
      "type": {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "label": "Cancelado",
    "value": "canceled",
  },
  {
    "color": "green",
    "icon": {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": null,
      "_store": {},
      "key": null,
      "props": {
        "size": 16,
        "strokeWidth": 3,
      },
      "type": {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
    "label": "Completado",
    "value": "completed",
  },
  {
    "color": "gray",
    "label": "Todos los estatus",
    "value": "",
  },
]
`;

exports[`SearchFilters should render with all filters enabled 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value=""
    />
    <input
      class="chakra-input css-b7g0y5"
      name="treatments"
      placeholder="Filtrar por servicio"
      value=""
    />
    <input
      class="chakra-input css-sjm060"
      name="date"
      placeholder="Filtrar por fecha"
      type="date"
      value=""
    />
    <div
      class="chakra-select__root css-3jlqt6"
      data-part="root"
      data-scope="select"
      dir="ltr"
      id="select::r0:"
    >
      <div
        class="chakra-select__control css-17j1053"
        data-part="control"
        data-scope="select"
        data-state="closed"
        dir="ltr"
        id="select::r0::control"
      >
        <button
          aria-controls="select::r0::content"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="select::r0::label"
          class="chakra-select__trigger css-12uc8un"
          data-part="trigger"
          data-placeholder-shown=""
          data-scope="select"
          data-state="closed"
          dir="ltr"
          id="select::r0::trigger"
          role="combobox"
          type="button"
        >
          Filtrar por estado
        </button>
        <div
          class="chakra-select__indicatorGroup css-1xvjckd"
        >
          <div
            aria-hidden="true"
            class="chakra-select__indicator css-1atg1ut"
            data-part="indicator"
            data-scope="select"
            data-state="closed"
            dir="ltr"
          >
            <svg
              class="css-s3mb0o"
              viewBox="0 0 24 24"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchFilters should render with date filter only 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value=""
    />
    <input
      class="chakra-input css-sjm060"
      name="date"
      placeholder="Filtrar por fecha"
      type="date"
      value="2023-01-01"
    />
  </div>
</div>
`;

exports[`SearchFilters should render with name filter filled 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value="John Doe"
    />
  </div>
</div>
`;

exports[`SearchFilters should render with status filter only 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value=""
    />
    <div
      class="chakra-select__root css-3jlqt6"
      data-part="root"
      data-scope="select"
      dir="ltr"
      id="select::r1:"
    >
      <div
        class="chakra-select__control css-17j1053"
        data-part="control"
        data-scope="select"
        data-state="closed"
        dir="ltr"
        id="select::r1::control"
      >
        <button
          aria-controls="select::r1::content"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="select::r1::label"
          class="chakra-select__trigger css-12uc8un"
          data-part="trigger"
          data-placeholder-shown=""
          data-scope="select"
          data-state="closed"
          dir="ltr"
          id="select::r1::trigger"
          role="combobox"
          type="button"
        >
          Confirmado
        </button>
        <div
          class="chakra-select__indicatorGroup css-1xvjckd"
        >
          <div
            aria-hidden="true"
            class="chakra-select__indicator css-1atg1ut"
            data-part="indicator"
            data-scope="select"
            data-state="closed"
            dir="ltr"
          >
            <svg
              class="css-s3mb0o"
              viewBox="0 0 24 24"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchFilters should render with treatment filter only 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-5578bu"
  >
    <input
      class="chakra-input css-b7g0y5"
      name="name"
      placeholder="Filtrar por nombre"
      value=""
    />
    <input
      class="chakra-input css-b7g0y5"
      name="treatments"
      placeholder="Filtrar por servicio"
      value="Haircut"
    />
  </div>
</div>
`;
